src/readout/ReadoutParserTest.cpp

```diff
@@ -0,0 +1,107 @@
+/** Copyright (C) 2016, 2017 European Spallation Source ERIC */
+
+#include <readout/ReadoutParser.h>
+#include <readout/ReadoutParserTestData.h>
+#include <test/TestBase.h>
+
+class ReadoutTest : public TestBase {
+protected:
+  ReadoutParser RdOut;
+  const int DataType{0x30};
+  void SetUp() override {}
+  void TearDown() override {}
+};
+
+TEST_F(ReadoutTest, Constructor) {
+  ASSERT_EQ(RdOut.Stats.ErrorBuffer, 0);
+  ASSERT_EQ(RdOut.Stats.ErrorSize, 0);
+  ASSERT_EQ(RdOut.Stats.ErrorVersion, 0);
+  ASSERT_EQ(RdOut.Stats.ErrorTypeSubType, 0);
+  ASSERT_EQ(RdOut.Stats.ErrorSeqNum, 0);
+}
+
+// nullptr as buffer
+TEST_F(ReadoutTest, ErrorBufferPtr) {
+  auto Res = RdOut.validate(0, 100, DataType);
+  ASSERT_EQ(Res, -ReadoutParser::EBUFFER);
+  ASSERT_EQ(RdOut.Stats.ErrorBuffer, 1);
+}
+
+// size is 0
+TEST_F(ReadoutTest, ErrorBufferSize) {
+  auto Res = RdOut.validate((char *)100, 0, DataType);
```

Comment:
Not only here, but in general? What is our "cast strategy"? Do we continue to use C-style casts,or switch to static_cast<char *>(100) ?

Reply:
Personally I hate the C++ style casts, but I am old. Luckily the LLVM code standard have no opinion on this. I have yet to find a really good example where there is an avantage to blablah_cast<>()